backend/mora/common.py

```diff
@@ -52,26 +53,48 @@ def get(self, objid):
 
 
 class RequestType(enum.Enum):
-    CREATE = 0
-    EDIT = 1
+    CREATE, EDIT, TERMINATE = range(3)
 
 
 # The handler mapping is populated by each individual active RequestHandler
-HANDLERS = {}
+HANDLERS_BY_ROLE_TYPE = {}
+HANDLERS_BY_FUNCTION_KEY = {}
+FUNCTION_KEYS = {}
 
 
-def register_request_handler(name):
-    """Decorator to register a writing request handler"""
-    def wrapper(cls):
-        HANDLERS[name] = cls
+class _RequestHandlerMeta(abc.ABCMeta):
```

Comment:
Using metaclasses seems unnecessarily complicated compared to using decorators as before, especially because now this code is run not only for the specific handler classes, but also for every abstract handler class as well.